src/Module/Xrd.php

```diff
@@ -65,13 +65,21 @@ protected function rawContent(array $request = [])
 
 		if (substr($uri, 0, 4) === 'http') {
 			$name = ltrim(basename($uri), '~');
+			$host = parse_url($uri, PHP_URL_HOST);
 		} else {
 			$local = str_replace('acct:', '', $uri);
 			if (substr($local, 0, 2) == '//') {
 				$local = substr($local, 2);
 			}
 
-			$name = substr($local, 0, strpos($local, '@'));
+			$pieces = explode('@', $local);
+			$name = $pieces[0];
+			$host = $pieces[1];
```

Comment:
Here's the PHP sugar to reduce the line count:
```suggestion
			list($name, $host)= explode('@', $local);
```

Reply:
I start to like php ... :)